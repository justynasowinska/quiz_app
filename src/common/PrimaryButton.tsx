import * as React from 'react';
import { Button } from 'react-native-elements';
import { StyleSheet } from 'react-native';

interface PropsType {
    title: string;
    onPress: () => void;
}

export const PrimaryButton = (props: PropsType) => (
    <Button
        title={props.title.toUpperCase()}
        onPress={props.onPress}
        buttonStyle={styles.buttonStyle}
    />
);

const styles = StyleSheet.create({
    buttonStyle: {
        paddingLeft: 20,
        paddingRight: 20,
        paddingTop: 10,
        paddingBottom: 10
    }
});