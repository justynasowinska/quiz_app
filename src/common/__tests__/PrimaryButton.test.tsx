import * as React from 'react';
import { shallow } from 'enzyme';

import { PrimaryButton } from '../PrimaryButton';

describe('PrimaryButton Component', () => {
    it('matches to the snapshot', () => {
        const component = shallow(
            <PrimaryButton
                title="Some title"
                onPress={jest.fn()}
            />
        );

        expect(component).toMatchSnapshot();
    });
});
