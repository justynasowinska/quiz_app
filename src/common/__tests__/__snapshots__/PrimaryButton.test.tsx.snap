// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton Component matches to the snapshot 1`] = `
<ForwardRef(Themed.Button)
  buttonStyle={
    Object {
      "paddingBottom": 22.5,
      "paddingLeft": 45,
      "paddingRight": 45,
      "paddingTop": 22.5,
    }
  }
  onPress={[MockFunction]}
  title="SOME TITLE"
  titleStyle={
    Object {
      "fontSize": 30,
    }
  }
/>
`;
